utils/server/openAiClient.ts

```diff
@@ -83,11 +112,36 @@ export const ChatCompletionStream = async (
   // HTTP POST error handling
   if (response.status !== 200) {
     const result = await response.json()
+    if (response.status === 401) {
+      if (result.error) {
+        throw new OpenAIAuthError(result.error.message)
+      }
+
+      throw new OpenAIAuthError(result.message)
+    }
+
+    if (response.status === 429 && result.error) {
+      const match = result.error.message.match(/retry after (\d+) seconds/)
+      const retryAfter = match ? parseInt(match[1]) : undefined
+      throw new OpenAIRateLimited(result.error.message, retryAfter)
+    }
+
+    if (result.error && result.error.code === "context_length_exceeded") {
+      const match = result.error.message.match(
+        /maximum context length is (\d+) tokens. However, you requested (\d+) tokens/
```

Comment:
Perhaps use the regex to make more resilient to change.
`/max.*length.* (\d+) tokens.*requested (\d+\) tokens/`